src/main/java/au/com/centrumsystems/hudson/plugin/buildpipeline/BuildPipelineView.java

```diff
@@ -606,11 +606,15 @@ private int triggerBuild(final AbstractProject<?, ?> triggerProject, final Abstr
         if (manualTrigger != null) {
             final Set<String> downstreamProjectsNames =
                     Sets.newHashSet(Splitter.on(",").trimResults().split(manualTrigger.getDownstreamProjectNames()));
-            if (downstreamProjectsNames.contains(project.getFullName())) {

```

Comment:
It also looks very bad for me, 'rerun' button and other internals knows about full item path and this search of project is bad.